lib/mongo.ex

```diff
@@ -319,6 +319,58 @@ defmodule Mongo do
     bangify(distinct(conn, coll, field, filter, opts))
   end
 
+
+  @doc """
+  Selects document a single document in a collection and returns either a document
+  or nil.
+
+  If multiple documents satisfy the query, this method returns the first document
+  according to the natural order which reflects the order of documents on the disk.
+
+  ## Options
+
+    * `:comment` - Associates a comment to a query
+    * `:cursor_type` - Set to :tailable or :tailable_await to return a tailable
+      cursor
+    * `:max_time` - Specifies a time limit in milliseconds
+    * `:modifiers` - Meta-operators modifying the output or behavior of a query,
+      see http://docs.mongodb.org/manual/reference/operator/query-modifier/
+    * `:cursor_timeout` - Set to false if cursor should not close after 10
+      minutes (Default: true)
+    * `:projection` - Limits the fields to return for all matching document
+    * `:skip` - The number of documents to skip before returning (Default: 0)
+  """
+  @spec find_one(conn, collection, BSON.document, Keyword.t) :: cursor
+  def find_one(conn, coll, filter, opts \\ []) do
+    query = [
+      {"$comment", opts[:comment]},
+      {"$maxTimeMS", opts[:max_time]}
+    ] ++ Enum.into(opts[:modifiers] || [], [])
+
+    query = filter_nils(query)
+
+    query =
+      if query == [] do
+        filter
+      else
+        filter = normalize_doc(filter)
+        filter = if List.keymember?(filter, "$query", 0), do: filter, else: [{"$query", filter}]
+        filter ++ query
+      end
+
+    select = opts[:projection]
+    opts = if Keyword.get(opts, :cursor_timeout, true), do: opts, else: [{:no_cursor_timeout, true}|opts]
+
+    drop = ~w(comment max_time modifiers sort cursor_type projection cursor_timeout order_by limit)a
+    opts = cursor_type(opts[:cursor_type]) ++ Keyword.drop(opts, drop)
+    opts = [{:limit, 1} | opts]
+
+    cursor = cursor(conn, coll, query, select, opts)
+    list = Enum.to_list(cursor)
+    List.first(list)
+  end
```

Comment:
I had also removed the modifier.  Not 100% sure how the modifier works.
      {"$orderby", opts[:sort]}

Enum.at(0) will return "0" if the cursor is empty, so I don't think "find" will work, so I think it must be it's own function.